terraform/templates/cloud-init/network_config.cfg

```diff
@@ -6,26 +6,15 @@ ethernets:
   interface${network_key}:
     match:
       macaddress: ${network_value.network_attachment.mac}
-%{ if network_value.network_info.mode=="nat" ~}
-    # This interface relies on DHCP because the virtualized Guest device
-    # is connected to a NAT and configured to claim only the requested IP
-    dhcp4: true
-    dhcp6: false
-%{ else ~}
-    # This interface is configured as STATIC because the related Guest virtualized network device
-    # is connected directly to a Host physical interface, so it can not claim an IP as intermediate.
-    # This way OS needs to claim the requested IP itself
-    dhcp4: false
-    dhcp6: false
-    addresses: [${network_value.network_attachment.address}]
-%{ endif ~}
+    addresses:
+      - ${network_value.network_attachment.address}
+%{ if length(networks) == 1 || network_value.network_attachment.default ~}
     routes:
-    - to: default # could be 0.0.0.0/0 optionally
-      via: ${network_value.network_info.gateway_address}
-      #metric: 100
-      on-link: true
```

Comment:
Why you delete this in all the cases?

Reply:
In this setup "on-link: true" seems that is not necessary. BTW, I'm not a networking expert and maybe I missing something.